### ISSUE TITLE
Not possible to create an Interface definition in synthetic AST

### ISSUE THREAD

**Issue by phated** [Microsoft/TypeScript#15497]
**TypeScript Version:**  2.3.2

**Code**

```ts
// A *self-contained* demonstration of the problem follows...

// There doesn't seem to be an API to create an Interface in the AST programmatically
const _implementsClause = ts.createHeritageClause(ts.SyntaxKind.ImplementsKeyword, [
    ts.createExpressionWithTypeArguments([], ts.createIdentifier('MyInterface'))
]);
const _exportModifier = ts.createToken(ts.SyntaxKind.ExportKeyword);
const _classExpression = ts.createClassExpression([_exportModifier], 'MyClass', [], [_implementsClause], []);
const _classStatement = ts.createStatement(_classExpression);

let src = ts.createSourceFile('output.ts', '', ts.ScriptTarget.ES5, false, ts.ScriptKind.TS);
src = ts.updateSourceFileNode(src, [_classStatement]);
```

**Expected behavior:**
There should be an API for creating an Interface definition in a synthetic AST.

**Actual behavior:**
Doesn't seem like the API exists.

**Comment by mhegazy** [comment]
You need to add a `createInterfaceDeclaration` to factory.ts, something a la [createClassDeclaration](https://github.com/Microsoft/TypeScript/blob/7080c4155b2c6cc69c5dae3f312d720d434da3a3/src/compiler/factory.ts#L1421-L1430).

**Comment by mhegazy** [comment]
PRs welcomed.

**Comment by phated** [comment]
@mhegazy will do!  One question though: are there tests around these APIs?  I couldn't find any.

**Comment by mhegazy** [comment]
No.

**Comment by mohsen1** [comment]
@phated can you also fix #15489 in your PR?

**Comment by phated** [comment]
I'll submit another one for that but I'm tackling some interface stuff right now.
